macro/src/lib.rs

```diff
@@ -33,25 +33,53 @@ struct AttrArgs {
     key_expr: Expr,
     store_type: Option<Type>,
     store_init: Option<Expr>,
+    result: Option<bool>,
+    option: Option<bool>,
```

Comment:
We feel that we'd like this feature only for `Result`.
Because... the only reason this needs to be a built-in feature and that it's not possible via the implementation of the trait is that most error types are not `Clone`.
And `Option` does not have an error value. So for `Option`, this is not necessary because it's possible to implement this behavior via the implementation of the trait.
Additionally, we expect that any serious, non-experimental/throw-away use of michie will be implementing the trait, because the provided implementations are too trivial. So, if the trait is implemented anyway, including this behavior is trivial.
And that's why we wouldn't like `Option`, only `Result` in this feature.

Reply:
Fine by me; it satisfies my usecase cleanly enough. It just seems to me you may have confusion from future users who won't understand this reasoning.